workspaces/ballerina/ballerina-visualizer/src/views/BI/FlowDiagram/index.tsx

```diff
@@ -1209,10 +1249,28 @@ export function BIFlowDiagram(props: BIFlowDiagramProps) {
         const noFormSubmitOptions = !options ||
             (
                 options?.closeSidePanel === undefined
-                && options?.updateLineRange === undefined
+                && options?.isChangeFromHelperPane === undefined
                 && options?.postUpdateCallBack === undefined
             );
 
+        if (
+            options?.isChangeFromHelperPane &&
+            selectedNodeRef.current?.codedata &&
+            !selectedNodeRef.current.codedata.isNew
+        ) {
+            const baseStartLine = selectedNodeRef.current.codedata.lineRange.startLine;
+            const safeOffset = Math.max(0, baseStartLine.offset - 1);
+            let targetLine = { ...baseStartLine, offset: safeOffset };
+
+            const nodeBefore = model ? getNodeBefore(selectedNodeRef.current, model.nodes) : undefined;
+            if (nodeBefore && nodeBefore.codedata.lineRange.endLine.line < targetLine.line) {
```

Comment:
I think this will break if a node is added inside a block node, like an IF node. Could @senithkay please check this logic again for block nodes as well?

```ballerina
        int a = 1;
        if(a>2){
            // + new node
        } // new var will be added here
```